M2/Macaulay2/m2/code.m2

```diff
@@ -100,24 +102,26 @@ editMethod String := filename -> (
 	  editor, " ", filename))
 EDIT = method(Dispatch => Thing)
 EDIT Nothing := arg -> (stderr << "--warning: source code not available" << endl;)
-EDIT Sequence := x -> ((filename,start,startcol,stop,stopcol,pos,poscol) -> (
+EDIT FilePosition := x -> (
+     filename := x#0; start := x#1;
      editor := getViewer("EDITOR", "emacs");
      if 0 != chkrun concatenate(
 	  if getenv "DISPLAY" != "" and editor != "emacs" then "xterm -e ",
 	  editor,
 	  " +",toString start,
 	  " ",
 	  filename
-	  ) then error "command returned error code")) x
+	  ) then error "command returned error code")
 editMethod Command := c -> editMethod c#0
 editMethod Function := args -> EDIT locate args
 editMethod Sequence := args -> (
-     editor := getViewer("EDITOR", "emacs");
-     if args === () 
-     then chkrun concatenate(
-	  if getenv "DISPLAY" != "" and editor != "emacs" then "xterm -e ",
-	  editor)
-     else EDIT locate args
+    if args === () then (
+	editor := getViewer("EDITOR", "emacs");
```

Comment:
You could save one layer of indentation with something like this:
```m2
editMethod Sequence := args -> if args =!= () then EDIT locate args else (
    editor := ...
```

Reply:
this code needs rewriting anyway so I'm not gonna bother at this point.